homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -326,20 +334,25 @@ async def async_refresh_data(self, now):
         self.request_in_progress = False
         async_dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
 
-    @staticmethod
-    def _extract_start_date(battery_info):
-        """Extract the server date from the battery response."""
-        try:
-            return battery_info.answer["BatteryStatusRecords"]["OperationDateAndTime"]
-        except KeyError:
-            return None
-
     async def async_get_battery(self):
         """Request battery update from Nissan servers."""
 
         try:
             # Request battery update from the car
             _LOGGER.debug("Requesting battery update, %s", self.leaf.vin)
+            start_date = None
+            try:
+                start_server_info = await self.hass.async_add_executor_job(
```

Comment:
Can we store this between updates so we don't have to fetch it each time?

Reply:
I don't think so. In my testing, if this call isn't made (with the following 1 second sleep) then the API just never returns anything, we retry ten times and then give up. I believe doing this closely mimics the behaviour of the official app.